content/content_scripts/popup.js

```diff
@@ -0,0 +1,194 @@
+        statusDiv[0].textContent = "Password set successfully!";
+        await updatePopupContent();
+      } else {
+        statusDiv[0].textContent = response.error;
+        cpasswordInput.value = "";
+        passwordInput.value = "";
+      }
+    } catch (error) {
+      console.error("Error registering:", error);
+      statusDiv[0].textContent = "An error occurred while registering";
+    }
+  });
+
+  // Event listener for login button
+  loginForm.addEventListener("submit", async (event) => {
+    event.preventDefault();
+    const pass1 = password1.value;
+
+    statusDiv[1].textContent = "";
+
+    try {
+      // Send logout request to background script
+      const response = await new Promise((resolve, reject) => {
+        chrome.runtime.sendMessage(
+          { action: "login", password: pass1 },
+          (response) => {
+            if (chrome.runtime.lastError) {
+              reject(chrome.runtime.lastError);
+            } else {
+              resolve(response);
+            }
+          }
+        );
+      });
+      if (response && response.success === true) {
+        statusDiv[1].textContent = "Logged in successfully!";
+        handleServiceWorkerInjection();
+        await updatePopupContent();
+      } else {
+        statusDiv[1].textContent = response.error;
+        passwordInput.value = "";
+      }
+    } catch (error) {
+      // console.error('Error logging in:', error);
+      statusDiv[1].textContent = "An error occurred while logging in.";
+    }
+  });
+
+  // Event listener for logout button
+  logoutForm.addEventListener("submit", async (event) => {
+    event.preventDefault();
+    const pass2 = password2.value;
+
+    statusDiv[2].textContent = "";
+
+    try {
+      // Send logout request to background script
+      const response = await new Promise((resolve, reject) => {
+        chrome.runtime.sendMessage(
+          { action: "logout", password: pass2 },
+          (response) => {
+            if (chrome.runtime.lastError) {
+              reject(chrome.runtime.lastError);
+            } else {
+              resolve(response);
+            }
+          }
+        );
+      });
+
+      if (response && response.success === true) {
+        statusDiv[2].textContent = "Logged out successfully!";
+        await updatePopupContent();
+      } else {
+        statusDiv[2].textContent = response.error;
+        passwordInput.value = "";
+      }
+    } catch (error) {
+      console.error("Error logging out:", error);
+      statusDiv[2].textContent = "An error occurred while logging out.";
+    }
+  });
+
+  //function for adding eventlistener to toggle if they exist
+  let socialChecked = false;
+  let gamingChecked = false;
+
+  socialToggle.addEventListener("change", function () {
+    socialChecked = this.checked;
+  });
+
+  gamingToggle.addEventListener("change", function () {
+    gamingChecked = this.checked;
+  });
+
+  const handleServiceWorkerInjection = async () => {
```

Comment:
this is not a good practice to use. what if we need to add another checkbox aur another type of rules to add. Our conditional statements will increase exponentially. 
are you still facing the issue with id?

Reply:
The issue with the I'd is solved

Reply:
lets use the design pattern we were using before then. This is not scalable

Reply:
send an array of enabled boxes to worker